### PR TITLE
Publish Error Prone compatibility matrix on website

### LINKED ISSUE #724: Document the minimum version of errorprone required
### Problem

I was attempting to run the latest version of the rules using the version of errorprone built in to bazel 5.2.1.  This failed because it appears the latest version only works with versions of errorprone > 2.20.0 (or so).  I was not able to find any documentation of that, but there are some closed issues where people had similar problems.  It may make sense to have a compatibility matrix or something prominent in the documentation.

### PR DESCRIPTION
**Pull request by Stephan202**
This resolves #724 and addresses a discussion in #935. The result can be seen [here](https://error-prone.picnic.tech/compatibility/).

Suggested commit message:
```
Publish Error Prone compatibility matrix on website (#938)

The new `website/generate-version-compatibility-overview.sh` script
tests all combinations, and stores the result in a Jekyll data file.

Resolves #724.
```

### CODE REVIEW

File: website/_data/compatibility.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# An overview of Error Prone Support releases, along with compatible Error
+# Prone releases. This data was generated by `generate-version-compatibility-overview.sh`.
```
Comment (Stephan202) [inline website/_data/compatibility.yml:2]: This script takes quite some time to run (and that'll go up due to it's quadratic nature :speak_no_evil:), so committing the result is the only realistic option.

File: website/compatibility.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   validates that the rules can be read by the targeted version of Error Prone,
+   proving that the serialization format is compatible.
+
+| Error Prone Support version | Compatible Error Prone versions |
+| --------------------------- | ------------------------------- |
+{% for release in site.data.compatibility.releases -%}
+| [{{ release.version }}](https://github.com/PicnicSupermarket/error-prone-support/releases/tag/v{{ release.version }}) | {%
+for version in release.compatible -%}
+[{{ version }}](https://github.com/google/error-prone/releases/tag/v{{ version }}){% unless forloop.last %}, {% endunless %}
+{%- endfor %} |
```
Comment (Stephan202) [inline website/compatibility.md:31]: The formatting is funny here, but that seemed necessary to make sure that a proper Markdown table is produced. Suggestions welcome.

File: website/compatibility.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+{% endfor %}
+
+[^1]: Note that this [does not prove][source-binary-compat] that the Error Prone Support and Error Prone versions are _binary_ compatible. This limitation does not appear to be an issue in practice.
+[^2]: The approach taken here may yield false negatives, because a reported incompatibility may merely be due to a test API incompatibility.
```
Comment (Stephan202) [inline website/compatibility.md:35]: Wrapping these lines breaks the formatting :shrug:

**Review by Stephan202 — COMMENTED** [review]
Some context.

File: website/compatibility.md (hunk 2 above)
Comment (Venorcis) [reply website/compatibility.md:31]: image

github doesn't really seem to properly display this? 🤔

Comment (Stephan202) [reply website/compatibility.md:31]: So yeah, this PR is a bit "fake": we don't currently deploy the website from the `master` branch, but instead from the `website` branch. Not all logic there has been properly cleaned up yet, and some of the changes there are really not acceptable on `master`; hence the split.

But... these changes in this PR are also part of that branch, so you can see the result [live on the internet](https://error-prone.picnic.tech/compatibility/).

**Review by Venorcis — APPROVED** [review]

**Review by rickie — APPROVED** [review]
Awesome work @Stephan202 , really cool script 🚀 !

Thanks for the review @Venorcis!

**Pull request merged**